display_list/display_list_builder.cc

```diff
@@ -612,6 +632,7 @@ void DisplayListBuilder::clipRect(const SkRect& rect,
 void DisplayListBuilder::clipRRect(const SkRRect& rrect,
                                    SkClipOp clip_op,
                                    bool is_aa) {
+  checkForDeferredSave();
```

Comment:
Perhaps move this into the else clause below because clipRect will already check for deferred save.